HW3/Task31/Extensions/TailExtension.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Task31.Extensions
+{
+    internal static class TailExtension<T> where T : struct
+    {
```

Comment:
This is not an extension method )) =>Create a universal extension method Tail<T>() for the IQueue<T> interface. This method returns a new queue consisting of the elements of the original parameter queue minus the first element.